File: pk/fib/fib.go
package main

import "fmt"

func fibonacci(n int) int {
	if n == 0 {
		return 0
	} else if n == 1 {
		return 1
	}
	return fibonacci(n-2) + fibonacci(n-1)
}

func main() {
	fmt.Println(fibonacci(40))
}

// go run fib.go  1.19s user 0.17s system 99% cpu 1.369 total

File: pk/fib/Fib.java
// Fib.java
public class Fib {
  public static void main(String[] args) {
    System.out.println(fibonacci(40));
  }
  public static int fibonacci(int n) {
    if(n == 0) {
      return 0;
    } else if(n == 1) {
      return 1;
    }
    return fibonacci(n - 2) + fibonacci(n - 1);
  }
}

// java Fib  0.76s user 0.05s system 91% cpu 0.880 total
